tests/test_refObjLoader.py

```diff
@@ -0,0 +1,289 @@
+# This file is part of meas_algorithms.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+import os.path
+import tempfile
+import unittest
+
+import numpy as np
+from smatch.matcher import sphdist
+import astropy.time
+
+import lsst.daf.butler
+import lsst.afw.geom as afwGeom
+from lsst.daf.butler import DatasetType, DeferredDatasetHandle
+from lsst.daf.butler.script import ingest_files
+from lsst.meas.algorithms import (ConvertReferenceCatalogTask, ReferenceObjectLoader)
+import lsst.utils
+import lsst.geom
+
+import ingestIndexTestBase
+
+
+class TestRefObjLoader(ingestIndexTestBase.ConvertReferenceCatalogTestBase,
+                       lsst.utils.tests.TestCase):
+    def setUp(self):
```

Comment:
It's a shame that this gets run for every test when seemingly it's a readonly butler that is created and the tests aren't multi-threaded (at least I don't see any of the tests writing to a butler since the butler is not stored anywhere apart from the deferred dataset refs). Otherwise there is a lot of overhead creating a butler and populating it and then doing it all again. Can you see if setUpClass works?

Reply:
Running all the tests here only takes 10s or so, even with the overhead.  I was having trouble with setupClass following the the same format of the other test this was based on.  There seemed to be some sort of filename clashes when running with xdist. The performance didn't seem like a problem and I didn't want to spend too much time trying to optimize.

Reply:
The problem might have been before I fixed all the temporary directory calls.  But once I got it working I didn't want to poke it any further...

Reply:
This change works for me:

```diff
diff --git a/tests/test_refObjLoader.py b/tests/test_refObjLoader.py
index 1941b559..4519d4e2 100644
--- a/tests/test_refObjLoader.py
+++ b/tests/test_refObjLoader.py
@@ -39,15 +39,18 @@ import ingestIndexTestBase
 
 class TestRefObjLoader(ingestIndexTestBase.ConvertReferenceCatalogTestBase,
                        lsst.utils.tests.TestCase):
-    def setUp(self):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
         np.random.seed(10)
 
         # Generate a second catalog, with different ids
         inTempDir = tempfile.TemporaryDirectory()
         inPath = inTempDir.name
-        skyCatalogFile, _, skyCatalog = self.makeSkyCatalog(inPath, idStart=25, seed=123)
+        skyCatalogFile, _, skyCatalog = cls.makeSkyCatalog(inPath, idStart=25, seed=123)
 
-        self.skyCatalog = skyCatalog
+        cls.skyCatalog = skyCatalog
 
         # override some field names, and use multiple cores
         config = ingestIndexTestBase.makeConvertConfig(withRaDecErr=True, withMagErr=True,
@@ -59,8 +62,8 @@ class TestRefObjLoader(ingestIndexTestBase.ConvertReferenceCatalogTestBase,
         config.file_reader.format = 'ascii.commented_header'
         config.n_processes = 1
         config.id_name = 'id'  # Use the ids from the generated catalogs
-        self.repoTempDir = tempfile.TemporaryDirectory()
-        repoPath = self.repoTempDir.name
+        cls.repoTempDir = tempfile.TemporaryDirectory()
+        repoPath = cls.repoTempDir.name
 
         # Convert the input data files to our HTM indexed format.
         dataTempDir = tempfile.TemporaryDirectory()
@@ -69,7 +72,7 @@ class TestRefObjLoader(ingestIndexTestBase.ConvertReferenceCatalogTestBase,
         converter.run([skyCatalogFile])
 
         # Make a temporary butler to ingest them into.
-        butler = self.makeTemporaryRepo(repoPath, config.dataset_config.indexer.active.depth)
+        butler = cls.makeTemporaryRepo(repoPath, config.dataset_config.indexer.active.depth)
         dimensions = [f"htm{depth}"]
         datasetType = DatasetType(config.dataset_config.ref_dataset_name,
                                   dimensions,
@@ -95,8 +98,8 @@ class TestRefObjLoader(ingestIndexTestBase.ConvertReferenceCatalogTestBase,
         for dataRef in datasetRefs:
             handles.append(DeferredDatasetHandle(butler=butler, ref=dataRef, parameters=None))
 
-        self.datasetRefs = datasetRefs
-        self.handles = handles
+        cls.datasetRefs = datasetRefs
+        cls.handles = handles
 
         inTempDir.cleanup()
         dataTempDir.cleanup()
@@ -270,10 +273,12 @@ class TestRefObjLoader(ingestIndexTestBase.ConvertReferenceCatalogTestBase,
         )
         self.assertEqual(result.fluxField, 'a_flux')
 
-    def tearDown(self):
+    @classmethod
+    def tearDownClass(cls):
+        super().tearDownClass()
         # Clear this to clean up the temporary directory.
-        self.repoTempDir.cleanup()
-        del self.repoTempDir
+        cls.repoTempDir.cleanup()
+        del cls.repoTempDir
 
 
 class TestMemory(lsst.utils.tests.MemoryTestCase):
```

and takes the non-multiprocessing testing from 16 seconds to 6 seconds and still works with xdist.